modules/programs/tmux.nix

```diff
@@ -0,0 +1,203 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.tmux;
+
+  mkChildPackageType = p: (
+   let
+     ptokens = splitString p.name "";
+     ptitle = toUpper (head ptokens) + concatStrings (tail ptokens);
+   in types.submodule {
+    options = {
+      enable = mkEnableOption "${ptitle}";
+
+      package = mkOption {
+        type = types.package;
+        default = p;
+        defaultText = "pkgs.${p.name}";
+        description = "Also install ${ptitle}.";
+      };
+    };
+  });
+
+  mkTmuxOption = name: opts: (types.submodule {
+    options = mkMerge [{
+      # FIXME: should be renamed.
+      specifier = mkOption {
+        type = types.string;
+        description = "Type of ${name}.";
+        default = "";
+        example = "\"-key\" becomes \"${name}-key\".";
```

Comment:
Hmm, I don't know anything about tmux. If I put `bind.specifier = ''"-key" becomes "bind-key"''` in my configuration tmux will do something sensible?